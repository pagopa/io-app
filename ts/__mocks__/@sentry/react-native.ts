export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
