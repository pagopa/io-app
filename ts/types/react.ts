import { Component, ComponentType, StatelessComponent } from "react";
import {
  NavigationInjectedProps,
  NavigationScreenProps
} from "react-navigation";

/**
 * Evaluates to the Props type of a React component
 */
export type ComponentProps<C> = C extends StatelessComponent<infer P1>
  ? P1
  : C extends Component<infer P2>
    ? P2
    : C extends ComponentType<infer P3> ? P3 : never;

/**
 * Infers the type of the navigation params of a component
 */
export type InferNavigationParams<
  C,
  P = ComponentProps<C>
> = P extends NavigationInjectedProps<infer N>
  ? N
  : P extends NavigationScreenProps<infer N1> ? N1 : never;
