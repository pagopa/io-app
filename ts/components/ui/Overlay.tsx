import * as React from "react";
import { StyleSheet, View } from "react-native";

const DEFAULT_OVERLAY_OPACITY = 1;
const DEFAULT_BACKGROUND_COLOR = "#fff";

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  overlay: {
    position: "absolute",
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    backgroundColor: "#fff",
    zIndex: 1,
    justifyContent: "center"
  },
  back: {
    zIndex: 0
  }
});

type Props = Readonly<{
  foreground?: React.ReactNode;
  opacity?: number;
  backgroundColor?: string;
}>;

/**
 * Creates a full screen overlay on top of another screen.
 *
 * Used for loading spinners and error screens.
 */
export const Overlay: React.SFC<Props> = props => {
  const {
    opacity = DEFAULT_OVERLAY_OPACITY,
    backgroundColor = DEFAULT_BACKGROUND_COLOR
  } = props;
  return (
    <View style={styles.container}>
      {props.foreground && (
        <View
          style={[
            styles.overlay,
            {
              opacity,
              backgroundColor
            }
          ]}
        >
          {props.foreground}
        </View>
      )}

      <View style={[styles.container, styles.back]}>{props.children}</View>
    </View>
  );
};
