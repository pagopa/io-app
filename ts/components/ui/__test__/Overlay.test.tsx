import * as React from "react";
import { render } from "@testing-library/react-native";
import { Text } from "react-native";
import { Overlay } from "../Overlay";

describe("Overlay", () => {
  it("Should match base snapshot", () => {
    const component = render(<Overlay />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("Should match all-properties snapshot", () => {
    const backgroundColor = "#FF0000";
    const opacity = 0.65;
    const foreground = <Text>This is a foreground</Text>;
    const children = <Text>This is a child</Text>;
    const component = render(
      <Overlay
        backgroundColor={backgroundColor}
        opacity={opacity}
        foreground={foreground}
      >
        {children}
      </Overlay>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
