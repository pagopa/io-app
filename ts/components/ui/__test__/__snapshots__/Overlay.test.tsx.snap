// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Should match all-properties snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="overlayComponent"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "bottom": 0,
          "inset": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "#FF0000",
          "opacity": 0.65,
        },
      ]
    }
  >
    <Text>
      This is a foreground
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "zIndex": 0,
        },
      ]
    }
  >
    <Text>
      This is a child
    </Text>
  </View>
</View>
`;

exports[`Overlay Should match base snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="overlayComponent"
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "zIndex": 0,
        },
      ]
    }
  />
</View>
`;
